### DISCUSSION TITLE
Recommended way of removing the date from the permalink?

### DISCUSSION THREAD

**Discussion by mendhak** [11ty/eleventy#2734] [Q&A] [answered]
Hi all,

Eleventy can get a post's date from the filename, for example, `2023-01-09-my-blog-post.md` and the date will be `2023-01-09`. So far so good.

However the permalink for this post becomes `/2023-01-09-my-blog-post/` and I would like it to be `/my-blog-post/`.
Can you recommend a way to remove the date part from the permalink when it's being generated?

Thanks
mendhak

**Comment by pdehaan** [comment, marked as answer]
You could create a directory data file which sets the `permalink` via an `eleventyComputed` function:
```js
module.exports = {
  eleventyComputed: {
    permalink(data) {
      if (data.permalink) {
        return data.permalink;
      }
      if (data.page.filePathStem.endsWith("/index")) {
        return `${ data.page.filePathStem }.html`;
      }
      return `${ data.page.filePathStem }/`;
    }
  }
};
```

```sh
tree -aI node_modules
.
├── .eleventy.js
├── package-lock.json
├── package.json
├── src/
│   └── pages/
│       ├── 2023-01-09-my-blog-post.md
│       ├── 2023-04-29-custom.md
│       ├── index.md
│       └── pages.11tydata.js
└── www/
    ├── hello/world/index.html
    └── pages/
        ├── index.html
        └── my-blog-post/index.html

7 directories, 10 files
```

**Reply by mendhak** [reply]
Aaah eleventyComputed, the one page in the docs I didn't read properly :laughing: 

Thanks for that, it's just what I need.  I tweaked it a bit and used page.fileSlug to get rid of the folder name too, awesome.

Also afterwards I played around and didn't realize you can put template stuff in json too.  

```json
{
  "permalink": "/{{ page.fileSlug }}/index.html",
}

```

Thanks so much, the features are quite powerful!
